pkg/resource/secret_reconciler.go

```diff
@@ -27,15 +27,16 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	"github.com/crossplaneio/crossplane-runtime/pkg/logging"
+	"github.com/crossplaneio/crossplane-runtime/pkg/util"
 )
 
 // Supported resources with all of these annotations will be fully or partially
 // propagated to the named resource of the same kind, assuming it exists and
 // consents to propagation.
 const (
-	AnnotationKeyPropagateToNamespace = "crossplane.io/propagate-to-namespace"
-	AnnotationKeyPropagateToName      = "crossplane.io/propagate-to-name"
-	AnnotationKeyPropagateToUID       = "crossplane.io/propagate-to-uid"
+	AnnotationKeyPropagateToNamespaces = "crossplane.io/propagate-to-namespace"
+	AnnotationKeyPropagateToNames      = "crossplane.io/propagate-to-name"
+	AnnotationKeyPropagateToUIDs       = "crossplane.io/propagate-to-uid"
```

Comment:
I wonder if instead of a comma separated list we could encode the different UIDs in the annotation keys. The namespace and name are really just there for human readability, so maybe something like:

`crossplane.io/propagate-to/668cd01a-17ab-11ea-8d71-362b9e155667: default/coolsecret`

Or maybe:

`to.propagate.crossplane.io/668cd01a-17ab-11ea-8d71-362b9e155667: default/coolsecret`

Reply:
I actually experimented with this a little bit and ended up finding it a little cleaner for a few reasons related to [annotation structure](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set):
- Only one slash can be used to separate the `prefix` from the `name` (we could obviously utilize another character for the separations, but it might look a little weird)
- The `name` is limited to 63 characters max (`668cd01a-17ab-11ea-8d71-362b9e155667:default/coolsecret` is already 55 chars so I think we could hit this quickly`)

Reply:
> I actually experimented with this a little bit and ended up finding it a little cleaner for a few reasons related to annotation structure

Do you mean it's a little clearer to stick with your proposed comma separated pattern?

Reply:
Yep, that's what I was indicating. Do you feel strongly in favor of the putting the secret in the annotation keys?

Reply:
So I did a little more reading about this. My read is that that the annotation key is of the format `optional-prefix/name`. The _name_ is limited to 63 characters, but the prefix is limited to 253. So I think something like `to.propagate.crossplane.io` would work.

https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/

I also think that if we switch the controller and predicate around to queue the reconcile for the _destination_ rather than the source secret we can avoid iterating over all of the annotations of the destination and instead:

1. Read the annotations of the destination secret.
1. Grab its source secret, and check whether the destination secret's UUID exists in its annotation keys.
1. If all seems well, copy the source data to the destination.